### PR TITLE
Refactor Helper functions and Zeeman for better performance

### PR DESCRIPTION
**Pull request by rpep**

### CODE REVIEW

File: fidimag/micro/zeeman.py [hunk 1]
```diff
@@ -129,20 +130,136 @@ def time_fun(pos, t):
...
+
+
+class SimpleTimeZeeman(Zeeman):
+
+    """
+    The time dependent external field, also can vary with space
+
+    The function time_fun must be a function which takes two arguments:
+
+    def time_fun(t):
```
Comment (davidcortesortuno) [inline fidimag/micro/zeeman.py]: `(t,pos)` ? or should tis depend on the mesh?

File: fidimag/micro/zeeman.py [hunk 2]
```diff
@@ -129,20 +130,136 @@ def time_fun(pos, t):
...
+        # (see the calculation in the *compute_energy* function)
+        self.Ms_long.shape = (3, -1)
+        for i in range(mesh.n):
+            self.Ms_long[:, i] = Ms[i]
+
+        self.Ms_long.shape = (-1,)
+        self.field = np.zeros(3 * self.n)
+
+    def compute_field(self, t=0, spin=None):
+        v = self.time_fun(t)
```
Comment (davidcortesortuno) [inline fidimag/micro/zeeman.py]: is there spatial dependence allowed here?

File: fidimag/micro/zeeman.py [hunk 3]
```diff
@@ -129,20 +130,136 @@ def time_fun(pos, t):
...
+
+    def compute_field(self, t=0, spin=None):
+        self.field[:] = helper.init_vector_func_fast(self.time_fun,
+                                                     self.mesh,
+                                                     False,
+                                                     t)
+        return self.field
+
+
+class SimpleTimeZeeman(Zeeman):
```
Comment (davidcortesortuno) [inline fidimag/micro/zeeman.py]: Did you test this function?

**Comment by rpep** [comment]
Think I've addressed all of your suggestions now David if you could have another look that'd be great!

File: fidimag/micro/zeeman.py (hunk 1 above)
Comment (rpep) [reply fidimag/micro/zeeman.py]: Fixed the documentation here

File: fidimag/micro/zeeman.py (hunk 3 above)
Comment (rpep) [reply fidimag/micro/zeeman.py]: Refactored this, it shouldn't have been used here.

**Pull request merged**